code/go/0chain.net/smartcontract/zcnsc/authorizer.go

```diff
@@ -216,3 +216,67 @@ func (zcn *ZCNSmartContract) DeleteAuthorizer(tran *transaction.Transaction, _ [
 
 	return response, err
 }
+
+func (zcn *ZCNSmartContract) UpdateAuthorizerConfig(_ *transaction.Transaction, inputData []byte, ctx cstate.StateContextI) (string, error) {
+	const (
+		code = "update_authorizer_settings"
+	)
+
+	gn, err := GetGlobalNode(ctx)
+	if err != nil {
+		msg := fmt.Sprintf("failed to get global node, err: %v", err)
+		err = common.NewError(code, msg)
+		Logger.Error("get global node", zap.Error(err))
+		return "", err
+	}
+
+	in := &AuthorizerNode{}
+	if err = in.Decode(inputData); err != nil {
```

Comment:
Could we use a [StringMap](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/config.go#L31) as used elsewhere. E.g. [faucet](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/faucetsc/sc.go#L106) or [interestpool](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/interestpoolsc/config.go#L43).

The problem with using an object rather than a map is how do you tell what fields you are changing and which you wish to remain as they are?